jupyter_server/services/events/handlers.py

```diff
@@ -75,6 +79,45 @@ def on_close(self):
         self.event_bus.handlers.remove(self.logging_handler)
 
 
+class EventHandler(APIHandler):
+    """REST api handler for events"""
+
+    auth_resource = AUTH_RESOURCE
+
+    @property
+    def event_bus(self):
+        """Jupyter Server's event bus that emits structured event data."""
+        return self.settings["event_bus"]
+
+    @web.authenticated
+    @authorized
+    async def post(self):
+        payload = self.get_json_body()
+        if payload is None:
+            raise web.HTTPError(400, "No JSON data provided")
+
+        try:
+            if "timestamp" in payload:
+                timestamp = datetime.strptime(payload["timestamp"], "%Y-%m-%d %H:%M:%S")
+            else:
+                timestamp = None
```

Comment:
Should we record local server time here?

Reply:
EventLog adds a timestamp by default, this is an override in case the client would like a timestamp different than the server local time.
https://github.com/jupyter/telemetry/blob/4c7bd069cc8432f92792376861b4746d7bbedb6a/jupyter_telemetry/eventlog.py#L231-L241

Reply:
Cheers!